zerver/webhooks/patreon/view.py

```diff
@@ -0,0 +1,116 @@
+        event_type = ":".join(filename.split("_"))
+        return {http_header_key: event_type}
+
+    return fixture_to_headers
+
+
+fixture_to_headers = get_http_headers_from_filename("HTTP_X_PATREON_EVENT")
+
+# called when new patron joined
+def members_create(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = f"New patron joined! :tada:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+# called when a patron edited their membership information
+# including updates on payment charging events
+def members_update(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = (
+        f"A patron just updated their membership. :notifications:\nPatrons in total: {patron_count}"
+    )
+
+    return body
+
+
+def members_delete(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = (
+        f"A patron just deleted their membership. :exclamation:\nPatrons in total: {patron_count}"
+    )
+
+    return body
+
+
+# called when a new patron joined through custom pledge
+def members_pledge_create(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = f"New patron joined through custom pledge! :tada:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+def members_pledge_update(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = f"A patron just updated their pledge. :notifications:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+def members_pledge_delete(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = f"A patron just deleted their pledge. :exclamation:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+EVENTS_FUNCTION_MAPPER: Dict[str, Callable[[Dict[str, Any]], Optional[str]]] = {
+    "members:create": members_create,
+    "members:update": members_update,
+    "members:delete": members_delete,
+    "members:pledge:create": members_pledge_create,
+    "members:pledge:update": members_pledge_update,
+    "members:pledge:delete": members_pledge_delete,
+}
+
+
+@webhook_view("Patreon")
+@has_request_variables
+def api_patreon_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: Dict[str, Any] = REQ(argument_type="body"),
+) -> HttpResponse:
+
+    event = validate_extract_webhook_http_header(request, "X_PATREON_EVENT", "Patreon")
+    assert event is not None
+
+    handler = EVENTS_FUNCTION_MAPPER.get(event)
+    if handler is None:
+        raise UnsupportedWebhookEventType(event)
+
+    body = handler(payload)
+    topic = "notifications"
```

Comment:
Do we think `patreon` would be a better default here?  That would feel more natural for putting this in a stream with other content.

Reply:
Yes I agree. Should have spotted that.